src/gsi/jfunc.f90

```diff
@@ -415,60 +414,62 @@ subroutine read_guess_solution(dirx,diry,mype)
-                     iadateg,nlatg,nlong,nsigg,nrcleng
-       jiterstart=0
+                     iadateg,nlatg,nlong,nsigg,nxval,nxrclen,nxsclen,nxpclen
          
 ! Let all tasks read gesfile_in to pick up bias correction (second read)
+       if(nxsclen+nxpclen > 0)then
+          if ( nxsclen==nsclen  .and. nxpclen == npclen) then
+             read(12,end=1236) (yhatsave_r4(i),i=nclen1+1,nclen-ntclen)
+             do i=nclen+1,nclen-ntclen
+                diry%values(i)=real(yhatsave_r4(i),r_kind)
+             end do
+          else
+             read(12)
+             if(mype == 0) then
+                write(6,*) 'READ_GUESS_SOLUTION:  INCOMPATABLE RADIANCE COMPONENT in GESFILE, gesfile_in'
+                write(6,*) 'READ_GUESS_SOLUTION:  nrclen: input,current=',nxsclen,nsclen,nxpclen,npclen
+                write(6,*) 'READ_GUESS_SOLUTION:  ignoring previous radiance guess'
+             endif
+          endif
+       else
+          if(mype == 0) write(6,*) ' No bias correction in restart file '
+       endif
 
 ! Loop to read input guess fields.  After reading in each field & level,
 ! scatter the grid to the appropriate location in the xhat and yhatsave
 ! arrays.
-       do k=1,nval_levs
-          read(12,end=1236) xhatsave_g,yhatsave_g
-          do kk=1,itotsub
-             i1=ltosi_s(kk); i2=ltosj_s(kk)
-             fieldx(kk)=xhatsave_g(i1,i2)
-             fieldy(kk)=yhatsave_g(i1,i2)
-          end do
+       do k=1,nval_levs_ens
+          if(mype == myper)then
+             read(12,end=1236) yhatsave_g
+             do kk=1,itotsub
+                i1=ltosi_s(kk); i2=ltosj_s(kk)
+                fieldy(kk)=yhatsave_g(i1,i2)
+             end do
+          end if
           i=(k-1)*latlon11 + 1
-          call mpi_scatterv(fieldx,ijn_s,displs_s,mpi_real4,&
-                   xhatsave_r4(i),ijn_s(mm1),mpi_real4,myper,mpi_comm_world,ierror)
           call mpi_scatterv(fieldy,ijn_s,displs_s,mpi_real4,&
                    yhatsave_r4(i),ijn_s(mm1),mpi_real4,myper,mpi_comm_world,ierror)
-       end do  !end do over nval_levs
+       end do  !end do over val_levs_ens`
 
-!      Read radiance and precipitation bias correction terms
-       if ( nrcleng==nrclen ) then
-          read(12,end=1236) (xhatsave_r4(i),i=nclen1+1,nclen),(yhatsave_r4(i),i=nclen1+1,nclen)
-       else
-          if(mype == 0) then
-             write(6,*) 'READ_GUESS_SOLUTION:  INCOMPATABLE RADIANCE COMPONENT in GESFILE, gesfile_in'
-             write(6,*) 'READ_GUESS_SOLUTION:  nrclen: input,current=',nrcleng,nrclen
-             write(6,*) 'READ_GUESS_SOLUTION:  ignoring previous radiance guess' 
-          endif
-          do i=nclen1+1,nclen
-             xhatsave_r4(i)=zero
-             yhatsave_r4(i)=zero
-          end do
-       endif
-       do i=1,nclen
-          dirx%values(i)=real(xhatsave_r4(i),r_kind)
+       do i=1,nclen1
           diry%values(i)=real(yhatsave_r4(i),r_kind)
        end do
+       success = .true.
```

Comment:
Success is set .true. here.  What about the .false. case?   Should success be set to .false. elsewhere in this routine if the input file is not successfully read?

Reply:
Interesting question.  I think this parameter can be removed.  If the read is unsuccessful, it should use the initial gradient for the search direction (the same as without trying to read the guess).  There are lines put into stdout if the read fails.

Reply:
OK.  I think we are good as long as the code writes a message to stdout when the read fails.   If the read fails would we want code execution to terminate?   That is, would this be a fatal error (NCO terminology)?